flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/table/endpoint/hive/HiveServer2EndpointVersion.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.endpoint.hive;
+
+import org.apache.flink.annotation.PublicEvolving;
+import org.apache.flink.table.gateway.api.endpoint.EndpointVersion;
+
+import org.apache.hive.service.rpc.thrift.TProtocolVersion;
+
+/** Mapping between {@link HiveServer2EndpointVersion} and {@link TProtocolVersion} in Hive. */
+@PublicEvolving
+public enum HiveServer2EndpointVersion implements EndpointVersion {
+    HIVE_CLI_SERVICE_PROTOCOL_V1(TProtocolVersion.HIVE_CLI_SERVICE_PROTOCOL_V1),
+
+    HIVE_CLI_SERVICE_PROTOCOL_V2(TProtocolVersion.HIVE_CLI_SERVICE_PROTOCOL_V2),
+
+    HIVE_CLI_SERVICE_PROTOCOL_V3(TProtocolVersion.HIVE_CLI_SERVICE_PROTOCOL_V3),
+
+    HIVE_CLI_SERVICE_PROTOCOL_V4(TProtocolVersion.HIVE_CLI_SERVICE_PROTOCOL_V4),
+
+    HIVE_CLI_SERVICE_PROTOCOL_V5(TProtocolVersion.HIVE_CLI_SERVICE_PROTOCOL_V5),
+
+    HIVE_CLI_SERVICE_PROTOCOL_V6(TProtocolVersion.HIVE_CLI_SERVICE_PROTOCOL_V6),
+
+    HIVE_CLI_SERVICE_PROTOCOL_V7(TProtocolVersion.HIVE_CLI_SERVICE_PROTOCOL_V7),
+
+    HIVE_CLI_SERVICE_PROTOCOL_V8(TProtocolVersion.HIVE_CLI_SERVICE_PROTOCOL_V8),
+
+    HIVE_CLI_SERVICE_PROTOCOL_V9(TProtocolVersion.HIVE_CLI_SERVICE_PROTOCOL_V9),
+
+    HIVE_CLI_SERVICE_PROTOCOL_V10(TProtocolVersion.HIVE_CLI_SERVICE_PROTOCOL_V10);
```

Comment:
Thanks for pointing out! After offline discussion, I think the only way to support HiveSever2 in Hive 3.x is to expose `getHiveServer2Endpoint` and `getHiveServer2EndpointVersion` in `HiveShim`. This is because HiveServer2Endpoint implements `TCLIService.Iface` and the interface is different in the different hive version.